### DISCUSSION TITLE
Adopt Project Lombok to reduce boiler plate code

### DISCUSSION THREAD

**Discussion by manticore-projects** [JSQLParser/JSqlParser#1244] [General]
Currently JSQL Parser uses a lot of Boiler Plate Code for the Getters/Setters.

Would we like to adopt [Project Lombok](https://projectlombok.org/features/all) which created this code automatically based on a few Annotations? Maybe after releasing 4.1 or 4.2 and switching to a 5.0?

I have written now a few new Statement Objects and found this boiler plate code quite annoying honestly.

**Comment by wumpz** [comment]
Sorry, but I do not like libraries that modify bytecode. Look into #581.

**Reply by manticore-projects** [reply]
Apologies, I was not aware that this has been discussed before.
I guess, everyone starting to work on JSQLParser and implementing new statements will come up with this very same idea eventually :-D

Maybe state it in the README, that Lombok or friends are not on the table.
